src/meta_schedule/measure_callback/update_cost_model.cc

```diff
@@ -42,7 +42,8 @@ class UpdateCostModelNode : public MeasureCallbackNode {
     pruned_candidate.reserve(n);
     pruned_runner_result.reserve(n);
     for (int i = 0; i < n; i++) {
-      if (!builder_results[i]->error_msg.defined() &&
+      if (!builder_results[i]->error_msg.defined() &&  //
+          !runner_results[i]->error_msg.defined() &&   //
```

Comment:
We don't need to filter out failed runs because they are expected to be used as negative examples in cost models, for example see [here](https://cs.github.com/apache/tvm/blob/b4d4b82dbb9be2e4d0954f9dfd8e1c46079b66ee/python/tvm/meta_schedule/cost_model/xgb_model.py#L472) for xgboost model usage. I suggest that we change it to check `Sum(...) > 0` only when runner's error message is not defined.

Reply:
Thanks for pointing it out. It make sense to me